hooks/post_gen_project.py

```diff
@@ -153,4 +153,17 @@ def remove_unrequested_plugin_examples():
 7. Consider customizing the rest of your plugin metadata for display on the napari hub: 
    https://github.com/chanzuckerberg/napari-hub/blob/main/docs/customizing-plugin-listing.md
 """
-    )
+
+{% if cookiecutter.github_repository_url == 'y' %}
+    # try to install and update pre-commit
+    try:
+        print("install pre-commit ...")
+        subprocess.run(["pip", "install", "pre-commit"], stdout=subprocess.DEVNULL)
```

Comment:
Is this running in a venv? If not then this is a little bit alarming to be installing into whatever env people use to generate these cookiecutters. For me this is `base` which i try to keep very clean.

Reply:
yep, I struggled with that a bit as well.  my reasoning was that A) this is off by default and B) `pre-commit` is about as much of a global application as `cookiecutter` is, so if we're in an environment where they installed cookiecutter, installing pre-commit is even a bit lighter.  

the alternative is to not offer the option of installing at all, and just expect them to install pre-commit themselves.  we can make that change if you think it should be that way